apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/Keys.cs

```diff
@@ -275,49 +255,56 @@ private V1.KeyRange ToProtobuf()
     /// </summary>
     public sealed class Key
     {
-        /// <summary>
-        /// The values of this key.
-        /// </summary>
-        internal ListValue KeyParts { get; }
+        internal SpannerParameterCollection Parameters { get; }
 
         /// <summary>
         /// Creates a key from the values in the parameter collection.
         /// </summary>
         /// <param name="keyParts">The values that define the key</param>
         internal Key(SpannerParameterCollection keyParts)
         {
-            KeyParts = ToListValue(keyParts);
+            Parameters = keyParts;
         }
 
         /// <summary>
         /// Creates a key consisting of the given key parts.
         /// </summary>
         /// <param name="keyParts">The values that define the key</param>
-        public Key(params object[] keyParts)
+        public Key(params object[] keyParts) : this(new SpannerParameterCollection(keyParts.Select(j => new SpannerParameter { Value = j })))
         {
-            var protobufValues = keyParts.Select(
-                v => SpannerDbType.FromClrType(v?.GetType())?.ToProtobufValue(v) ?? Value.ForNull());
-            KeyParts = new ListValue { Values = { protobufValues } };
         }
 
-        /// <inheritdoc/>
-        public override bool Equals(object other)
-            => other is Key key && KeyParts.Equals(key.KeyParts);
+        /// <summary>
+        /// Gets the values of this Key.
+        /// </summary>
+        /// <param name="options">The options for conversion between protobuf and CLR types.</param>
+        /// <returns>The values of this Key.</returns>
+        internal ListValue GetKeyParts(SpannerConversionOptions options) => ToProtobuf(options);
```

Comment:
We don't need this method that does the same as T`oProtobuf`. I think the better name is `ToProtobuf`, or better still change the `ToProtobuf` in all 3 classes to `GetProtobuf`.